src/Features/faker/Faker.tsx

```diff
@@ -0,0 +1,369 @@
+import classes from "./styles.module.css";
+import {
+  ActionIcon,
+  Button,
+  Group,
+  ScrollArea,
+  Select,
+  Stack,
+  TextInput,
+  Divider,
+  rem,
+  Text,
+  NumberInput
+} from "@mantine/core";
+import { FakerInput } from "./FakerInput"
+import { useCallback, useEffect, useState } from "react";
+import { faker } from "@faker-js/faker";
+import { Monaco } from "../../Components/MonacoWrapper";
+import {
+  MdOutlineRemove,
+  MdOutlineClose,
+  MdOutlineWarning,
+  MdOutlineCheck,
+} from "react-icons/md";
+import { notifications } from "@mantine/notifications";
+import { allLocales } from "@faker-js/faker";
+import YAML from "js-yaml";
+import { delimiter } from "@tauri-apps/api/path";
+import { event } from "@tauri-apps/api";
+import { assert } from "quicktype-core";
+
+const errorIcon = <MdOutlineClose style={{ width: rem(20), height: rem(20) }} />;
+const warningIcon = <MdOutlineWarning style={{ width: rem(20), height: rem(20) }} />;
+
+/**
+ * Supported output formats
+ */
+const outputFormats = [
+  { format: "json", label: "JSON" },
+  { format: "yaml", label: "YAML" },
+  { format: "sql", label: "SQL" },
+  { format: "csv", label: "CSV" },
+];
+
+/**
+ * Generates a mock data using @faker-js/faker based on given data category and type
+ * 
+ * @param category The category for which the fake data falls within
+ * @param dataType The specific data type to be fakes
+ * @param locale The locale to be used for generating mock data
+ * @returns The faker generated data
+ */
+const getMockData = (category: string, dataType: string, locale: string | null): any => {
+  const actualLocale = locale || 'en_US'; // TODO: Can't seem to get faker to work
+  if ((faker as any)[category] && typeof (faker as any)[category][dataType] === 'function') {
```

Comment:
AFAIK, it doesn't (VS Code didn't suggest any). Also, the typed package I found was deprecated. I'll review before next week hopefully.

Reply:
So, I just had time to review your comment. The reason I used `any` was because I couldn't index `Faker` class (no index signature). Hence, I had to cast it to `any` to enable me to use the indexes as it is essential to the dynamic generation of fake data based on `Faker` definitions (categories) and chosen data type. 

If you can figure out a work-around, please go ahead and solve the issue.